dppl/sycl_core.pyx

```diff
@@ -72,15 +106,101 @@ cdef extern from "dppl_sycl_queue_interface.h":
                                              size_t device_num) except +
     cdef void DPPLSetAsDefaultQueue (_device_type DTy,
                                      size_t device_num) except +
-    cdef void DPPLDeleteQueue (DPPLSyclQueueRef Q) except +
 
-# Destructor for a PyCapsule containing a SYCL queue
-cdef void delete_queue (object cap):
-    DPPLDeleteQueue(<DPPLSyclQueueRef>PyCapsule_GetPointer(cap, NULL))
 
+cdef class SyclContext:
+    cdef DPPLSyclContextRef ctxt_ptr
```

Comment:
Previously it was PyCapsule for storing queue. It was singleton. Now extension classes store this pointer. It is singleton as before or we could have several now? What is lifetime of this object in the back of the pointer (when created, when deleted)?